### PR TITLE
Bug fix/optimize shard distribution api and ui

### PR DESCRIPTION
**Pull request by hkernbach**

### CODE REVIEW

**Comment by hkernbach** [comment]
**DO NOT** merge this until mchacki's approval.

File: arangod/Cluster/ShardDistributionReporter.cpp [hunk 1]
```diff
@@ -424,10 +418,59 @@ void ShardDistributionReporter::getDistributionForDatabase(
...
+      ReportInSync(col.get(), allShards.get(), aliases, result);
+    } else {
+      todoSyncStateCheck.push(col);
+    }
+  }
+
+  const bool progress = false;
+  helperDistributionForDatabase(dbName, result, todoSyncStateCheck, endtime, aliases, progress);
   result.close();
 }
```
Comment (mchacki) [inline arangod/Cluster/ShardDistributionReporter.cpp]: Is it possible to combine this two functions?
The forLoop + queue mgmt is identical.
Maybe we can get sth that takes the queue + the LogicalCollection as input.
@hkernbach Easier to explain in Person if you have questions

Comment (hkernbach) [reply arangod/Cluster/ShardDistributionReporter.cpp]: ok, lets discuss tomorrom. I'm not sure:
1. uses col as string
2. uses cols as vector etc.

Comment (hkernbach) [reply arangod/Cluster/ShardDistributionReporter.cpp]: no need to change it here.
approved by @mchacki

**Pull request merged**